### DISCUSSION TITLE
Issue with Registration token

### DISCUSSION THREAD

**Discussion by BlindHoneyBadger** [mondoohq/cnquery#4032] [General]
Hi i have a 11.2.0 client installed on a host.

When i attempt to register it with Mondoo console i get this error

Error: failed to log in client: failed to log in client: failed to do request: Post "https://eu.api.mondoo.com/AgentManager/RegisterAgent": read tcp (IP obscured):65262 -> 34.102.168.217:443: wsarecv: An existing connection was forcibly closed by the remote host.

I see that the token is valid, and on the client, when i run the "cnspec login --token" command i see that it is able to read the token on the console - as it retrieves the token description and expiry date.

What am i missing here?

**Comment by chris-rock** [comment]
Hi @BlindHoneyBadger Can you share the the full output of `cnspec register` command as well as `cnspec status`?

**Reply by BlindHoneyBadger** [reply]
Hi @chris-rock 
Thanks for the reply, much appreciated

PS C:\Users\xxx> cnspec status
WRN can't find any paths for providers, none are configured home-path="C:\\Users\\xxx\\.config\\mondoo\\provider
s" system-path="C:\\ProgramData\\Mondoo\\providers"
- no Mondoo configuration file provided, using defaults
ERR could not check upstream health error="failed to do request: Post \"https://us.api.mondoo.com/Health/Check\": read t
cp xxx:49995->34.160.242.34:443: wsarecv: An existing connection was forcibly closed by the remote host."
Error: failed to get latest version: error fetching latest release: Get "https://releases.mondoo.com/cnquery/latest.json
?ignoreCache=1": read tcp xxxx:49996->34.110.159.213:443: wsarecv: An existing connection was forcibly closed
by the remote host.

PS C:\Users\xxxc> cnspec register
WRN can't find any paths for providers, none are configured home-path="C:\\Users\\xxx\\.config\\mondoo\\provider
s" system-path="C:\\ProgramData\\Mondoo\\providers"
- no Mondoo configuration file provided, using defaults
WRN could not initialize certificate authentication error="agent credentials must be set"


So, it seems i am missing agent credentials. How do i add those?

**Reply by chris-rock** [reply]
Ah, I guess we should make the `cnspec register` command more intuitive by indicating that a token is required instead of showing the error.

```
cnspec register --token <token>
```

1. You need a registration from Mondoo platform, in case you have no access you can book a meeting here https://mondoo.com/pricing
2. It is strange that you see the TLS connection errors. Can you check that your clock is set correct?

The full documentation is available here https://mondoo.com/docs/cnspec/cnspec-adv-install/registration/

**Reply by BlindHoneyBadger** [reply]
I have access to the platform, i created the registration token there.

I tried registering by using this cnspec register --token "..........xNjE4ODQzNywib3duZXIiOiIiLCJzY29wZSI6Ii8vY2FwdGF............", but that gives me the error message which i pasted in my first post. 

The time on the client is correct, client has NTP servers access, and there is no time skew.

**Reply by chris-rock** [reply]
We cannot reproduce that issue. Registration with EU region works as expected in our cases. Please write an email to support@mondoo.com so we can help you better.
